interpreter/src/objects.rs

```diff
@@ -143,19 +141,28 @@ pub trait TryIntoValue {
     fn try_into_value(self) -> Result<Value, Self::Error>;
 }
 
-impl<T: Serialize> TryIntoValue for T {
-    type Error = SerializationError;
+#[cfg(feature = "serde")]
+impl<T: serde::Serialize> TryIntoValue for T {
+    type Error = crate::ser::SerializationError;
     fn try_into_value(self) -> Result<Value, Self::Error> {
-        to_value(self)
+        crate::ser::to_value(self)
     }
 }
-
+#[cfg(feature = "serde")]
 impl TryIntoValue for Value {
     type Error = Infallible;
     fn try_into_value(self) -> Result<Value, Self::Error> {
         Ok(self)
     }
 }
+#[cfg(not(feature = "serde"))]
+impl<T: Into<Value>> TryIntoValue for T {
+    type Error = Infallible;
+
+    fn try_into_value(self) -> Result<Value, Self::Error> {
+        Ok(self.into())
+    }
+}
```

Comment:
Which was required to implement `TryIntoValue for T: Into<Value>` when serde is disabled.

Aaandd.. that was required because disabling serde broke examples which used simple numbers.